base/multidimensional.jl

```diff
@@ -401,35 +401,38 @@ module IteratorsMD
         iterfirst, iterfirst
     end
     @inline function iterate(iter::CartesianIndices, state)
-        valid, I = __inc(state.I, iter.indices)
+        valid, I = __inc(state.I, iter.indices, Val(ndims(iter)))
         valid || return nothing
         return CartesianIndex(I...), CartesianIndex(I...)
     end
 
     # increment & carry
     @inline function inc(state, indices)
-        _, I = __inc(state, indices)
+        _, I = __inc(state, indices, Val(length(state)))
         return CartesianIndex(I...)
     end
 
     # Unlike ordinary ranges, CartesianIndices continues the iteration in the next column when the
     # current column is consumed. The implementation is written recursively to achieve this.
     # `iterate` returns `Union{Nothing, Tuple}`, we explicitly pass a `valid` flag to eliminate
     # the type instability inside the core `__inc` logic, and this gives better runtime performance.
-    __inc(::Tuple{}, ::Tuple{}) = false, ()
-    @inline function __inc(state::Tuple{Int}, indices::Tuple{OrdinalRangeInt})
+    __inc(::Tuple{}, ::Tuple{}, ::Val) = false, ()
+    @inline function __inc(state::Tuple{Int}, indices::Tuple{OrdinalRangeInt}, ::Val{N}) where {N}
         rng = indices[1]
         I = state[1] + step(rng)
-        valid = __is_valid_range(I, rng) && state[1] != last(rng)
+        if N == 1
```

Comment:
I'm not sure if I get the idea -- when will `N != 1` here?

Reply:
It's also be called at R435 as the last input `ndim` is passed deeper without any change.

BTW, all the test failures should be invalid state. Since we have
```julia
julia> iterate(1:2:typemax(Int), typemax(Int)-1)
(-9223372036854775808, -9223372036854775808)
```
I think it's OK to replace them?